src/Inline/Element/AbstractInline.php

```diff
@@ -22,4 +24,36 @@
      * Used for storage of arbitrary data
      */
     public $data = [];
+
+    /**
+     * @var AbstractInline|null
+     */
+    protected $parent;
+
+    /**
+     * @return AbstractBlock|AbstractInline|null
+     */
+    public function getParent()
+    {
+        return $this->parent ? : null;
```

Comment:
You could probably remove the `? : null` part, since `null` is the only value which would trigger that part of the expression anyway.